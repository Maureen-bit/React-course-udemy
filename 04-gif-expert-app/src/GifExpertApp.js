import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory.js";
import { GifGrid } from "./components/GifGrid.js";

const GifExpertApp = ({ defaultCategories = [] }) => {
    /* Enviamos como argumento una categoría por defecto para poder probar en el test ya que
    no hay manera de poner un state en las pruebas ya que cada estado no tiene un identificador*/
    const [categories, setCategories] = useState(defaultCategories);
    
    return(
        <>
            <h2> GifExpertApp </h2>
            <AddCategory setCategories={setCategories}/>
            <hr />
            <ol>
                {
                    /* No usar el indice como key para los elementos ya que es muy volátil y puede cambiar
                    en cualquier momento */
                    categories.map( category =>
                        (
                            <GifGrid 
                                category={category}
                                key={category}
                            />
                        )
                    )
                }
            </ol>
        </>
    )
};
 export default GifExpertApp;