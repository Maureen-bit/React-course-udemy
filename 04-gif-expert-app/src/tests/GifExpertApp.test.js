import React from 'react';
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Pruebas unitarias en el componente <GifExpertApp />", () => {
    test('Debe renderizar correctamente el componente', () => {
        let view = shallow( <GifExpertApp />);
        expect(view).toMatchSnapshot();
    });
    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['The Simpsons', 'Heidi'];
        let view = shallow( <GifExpertApp defaultCategories={categories} />);
        expect(view).toMatchSnapshot();
        expect(view.find('GifGrid').length).toBeGreaterThan(0);
    });
    
})