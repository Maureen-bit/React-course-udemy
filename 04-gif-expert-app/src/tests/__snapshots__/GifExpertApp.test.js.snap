// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias en el componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="The Simpsons"
      key="The Simpsons"
    />
    <GifGrid
      category="Heidi"
      key="Heidi"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas unitarias en el componente <GifExpertApp /> Debe renderizar correctamente el componente 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
